components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,138 @@
+import { useUserInfoQuery } from '../graphql/index'
+import { useRouter } from 'next/router'
+import _ from 'lodash'
+import styles from '../scss/profileDropDown.module.scss'
+
+type ProfileDropDownMenuProps = {
+  username: string
+  isAdmin: boolean
+}
+
+type CustomToggleProps = {
+  children?: React.ReactNode
+  className?: string | undefined
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {}
+}
+
+const ProfileDropdownMenu: React.FC<ProfileDropDownMenuProps> = ({
+  username,
+  isAdmin
+}) => {
+  const { data } = useUserInfoQuery({
+    variables: { username },
+    skip: !username
+  })
+  const router = useRouter()
+  const location = router.asPath
+
+  const fullname = _.get(data, 'userInfo.user.name', '')
+
+  const userInfo: UserInfo = {
+    // 'A' stands for Anonymous, in case user did not put in full name
+    username,
+    firstName: fullname.split(' ')[0] || 'A',
+    lastName: fullname.split(' ')[1] || ' ',
+    discordUserId: data?.userInfo?.user?.discordUserId ?? '',
+    discordUsername: data?.userInfo?.user?.discordUsername ?? '',
+    discordAvatarUrl: data?.userInfo?.user?.discordAvatarUrl ?? ''
+  }
+
+  const dropdownAdminMenuItems = [
+    { title: 'Lessons', path: '/admin/lessons' },
+    { title: 'Users', path: '/admin/users' },
+    { title: 'Alerts', path: '/admin/alerts' }
+  ]
+
+  const isActive = (path: string) => {
+    if (path === location) return `${styles['active']}`
+    return ''
+  }
+
+  const adminDropdownMenu = dropdownAdminMenuItems.map(({ title, path }) => (
+    <Dropdown.Item
+      className={`${styles['dropdown-item']} nav-link `}
+      href={path}
+      key={title}
+      bsPrefix={isActive(path)}
+    >
+      {title}
+    </Dropdown.Item>
+  ))
+
+  const buttonMenuToggle = React.forwardRef(
+    (props: CustomToggleProps, ref: React.Ref<HTMLDivElement>) => (
+      <div
+        ref={ref}
+        className={props.className}
+        onClick={e => {
+          props.onClick && props.onClick(e)
+        }}
+      >
+        {props.children}
+      </div>
+    )
+  )
+
+  return (
+    <Dropdown>
+      <div className="d-none d-lg-block">
+        <Dropdown.Toggle
+          as={buttonMenuToggle}
+          className={`${styles['nav-user-toggle']} `}
+          id="user_nav_toggle"
+        >
+          <UserInfoImage user={userInfo} className={`${styles['user-icon']}`} />
+          <span>{_.capitalize(username)}</span>
+          <ChevronDownIcon size={16} />
+        </Dropdown.Toggle>
+        <Dropdown.Menu className={`${styles['dropdown-menu']}`} align="end">
+          {isAdmin && (
+            <>
+              <Dropdown.Header>ADMIN</Dropdown.Header>
+              {adminDropdownMenu}
+              <Dropdown.Divider />
+            </>
+          )}
+          <Dropdown.Item
+            className={`${styles['dropdown-item']} nav-link `}
+            bsPrefix={isActive(`/profile/${username}`)}
+            href={`/profile/${username}`}
```

Comment:
There should be a constant made for `/profile`, seems like this string is repeated  a lot in this file

Reply:
Thanks! Fixed it!